spotifyutils/__init__.py

```diff
@@ -49,8 +46,49 @@ def cli(input_args=None):
     )
 
     args = parser.parse_args(input_args)
-    main(**vars(args))
+    configuration(**vars(args))
+
+def configuration(**kwargs):
+    """ Interpret CLI args and create configuration """
+    
+    # if configfile arg is passed, check if any other args are passed. If additional args are passed, overwrite any existing args 
+    # specified in configfile. If additional arguments are NOT passed, simply read the values stored in the configfile. 
+    # If configfile is not passed, just use CLI args passed.
+
+    if kwargs['configfile']:
+        configfile = kwargs['configfile']
+        config = configparser.ConfigParser()
+        if kwargs['client_id'] or kwargs['client_secret'] or kwargs['redirect_uri']:
+            config['spotifyutils'] = {}
+
+            for key, value in kwargs.items():
+                config['spotifyutils'][key] = value
+
+            with open(configfile, "w") as write_config:
+                config.write(write_config)
+            
+            client_id = kwargs['client_id']
+            client_secret = kwargs['client_secret']
+            redirect_uri = kwargs['redirect_uri']
+        else:
+            config.read(configfile)
+            client_id = config.get('spotifyutils', 'client_id')
+            client_secret = config.get('spotifyutils', 'client_secret')
+            redirect_uri = config.get('spotifyutils', 'redirect_uri')
+    else:
+        client_id = kwargs['client_id']
+        client_secret = kwargs['client_secret']
+        redirect_uri = kwargs['redirect_uri']
+
+    main(client_id, client_secret, redirect_uri)
```

Comment:
Calling a function named "main" here is confusing. This appears to be a configuration function, but it is doing much more than that with this call.